cl/corpus_importer/import_columbia/columbia_utils.py

```diff
@@ -766,3 +770,40 @@ def map_opinion_types(opinions=None) -> None:
             op["type"] = "040dissent"
         elif op_type and op_type == "concurrence":
             op["type"] = "030concurrence"
+
+
+def validate_citations(cites: list[str], cluster_id: int) -> list:
+    """Get valid citations
+
+    :param cites: citation list
+    :param cluster_id: cluster id related to citations
+    :return: list with valid citations
+    """
+
+    valid_citations = []
+
+    for cite in cites:
+        clean_cite = re.sub(r"\s+", " ", cite)
+        citation = get_citations(clean_cite)
+        if (
+            not citation
+            or not isinstance(citation[0], FullCaseCitation)
+            or not citation[0].groups.get("volume", False)
+        ):
+            logger.warning(f"Citation parsing failed for {clean_cite}")
+            continue
+
+        if Citation.objects.filter(
+            cluster_id=cluster_id, reporter=citation[0].corrected_reporter()
+        ).exists():
+            # Avoid adding a citation if we already have a citation from the
+            # citation's reporter
+            logger.info(
+                f"Cluster id: {cluster_id} - Reporter's citation already exist in "
```

Comment:
you added this whole function - but does it not repeat again the same code in the add_citations_to_cluster cluster?  

instead of just moving this around I think we just need to add a simple log.

Reply:
That's a very reasonable idea. I'd suggest checking the data to make sure there's not some insane edge case we haven't considered, but assuming there's not, it's the right way to solve this. Don't forget to include a multi-column index when you do!